fix: TASK-#15126 Airtable queries able to show the total count of rec…

Naveen-Goud opened the pull request:
**Description**
-This is the PR to show the total count of records for airtable queries. 

fixes: [#15126](https://github.com/appsmithorg/appsmith/issues/15126) 

 **changes in PR**
 -Updated the file in frontend file QueryDebuggerTabs.tsx, to make the data that is coming from api into array of objects.
 -Updated the Frontend code , as a solution where the updated code will
       - will check if the response is array of objects or not
       - if not , then it will convert into an array of objects.

**difference between PostgreSQL and Airtabls:**
The main difference between relational and non-relational databases lies in how they represent data. Relational databases, like PostgreSQL, typically return data as an array of objects. This format is straightforward, with each object representing a row from a database table. On the other hand, non-relational databases, such as Airtable, often return data in a more nested structure. For instance, Airtable's data might be returned as an object that contains an array of records, where each record is itself an object. This difference in data representation can lead to inconsistencies in how data is processed and displayed, especially when integrating with systems expecting a specific format. Understanding and handling these differences is crucial for seamless data integration and manipulation across different types of databases

![Screenshot from 2024-06-25 12-55-41](https://github.com/zemoso-int/appsmith-from-the-business/assets/72243823/61a8ebd0-c719-41ef-a7a9-aa463aa5f466)
![Screenshot from 2024-06-25 12-55-49](https://github.com/zemoso-int/appsmith-from-the-business/assets/72243823/c62ff8f1-3c4d-414a-89fc-473d8605f93d)

- We can modify the SaasPlugin backend code to standardize API responses as arrays of objects. However, this is complex due to the need to handle various SaaS integrations like Google Sheets and AWS Lambda, requiring significant changes to ensure consistent data formatting across all services.

akshayvijayjain commented on app/client/src/pages/Editor/QueryEditor/QueryDebuggerTabs.tsx:
```diff
@@ -162,7 +162,15 @@ function QueryDebuggerTabs({
       }
     } else {
-      output = actionResponse.body as any;
+      if (typeof actionResponse.body === "object" && actionResponse.body !== null) {
+        if (Array.isArray(actionResponse.body)) {
+          output = actionResponse.body;
+        } else if ('records' in actionResponse.body && Array.isArray((actionResponse.body as any).records)) {
+          output = (actionResponse.body as any).records;
+        } else {
+          output = [actionResponse.body];
+        }
+      }
```
@Naveen-Goud Thank you for changes.

Please update the code as per coderabbit suggestion and do comment if the suggestion is invalid.

Naveen-Goud commented:
I have reviewed and addressed the comments received from CodeRabbit. To resolve the issues, I added a separate method to handle the response parsing, thereby avoiding code duplication. After making these changes,

I cross-checked everything to ensure it works as expected. As a result, we can safely ignore the latest suggestion from CodeRabbit.ai.

akshayvijayjain commented on app/client/src/pages/Editor/QueryEditor/QueryDebuggerTabs.tsx:
```diff
@@ -68,9 +68,9 @@ const parseResponseBody = (body: any): any[] => {
     return parsedOutput;
   } else if (parsedOutput.records && Array.isArray(parsedOutput.records)) {
```
we can remove  else from the line 69 also

akshayvijayjain reviewed (approved):
we are doing this changes on frontend only.

because to handle this change on backend means to have same response returned by SAAS DB and pgsql will required to work on many DB plugins.

Naveen-Goud commented:
Hii @ajinkyakulkarni and @chandannkumar, Please review this PR?

Naveen-Goud commented:
hii @rohan-arthur @Aishwarya-U-R @sumitsum , I hope you're well. I’m curious and excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!

Thank you.

Naveen-Goud commented:
hii @rohan-arthur @Aishwarya-U-R @sumitsum , I hope you're well. I’m curious and excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!

Thank you.

Naveen-Goud commented:
hii @rohan-arthur @Aishwarya-U-R @sumitsum , I excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!

Thank you.

Naveen-Goud commented:
hii @rohan-arthur @Aishwarya-U-R @sumitsum @Nikhil-Nandagopal  @NilanshBansal , can you assign reviewer to this PR.

Thank you.
